digi/xbee/packets/zigbee.py

```diff
@@ -0,0 +1,360 @@
+    and the key entry will never expire unless explicitly removed.
+
+    To remove a key entry on a distributed trust center, this frame should be
+    issued with a null (None) key. In a  centralized trust center you cannot
+    use this method to explicitly remove the key entries.
+
+    .. seealso::
+       | :class:`.XBeeAPIPacket`
+    """
+
+    __MIN_PACKET_LENGTH = 17
+
+    def __init__(self, frame_id, registrant_address, options, key):
+        """
+        Class constructor. Instantiates a new :class:`.RegisterJoiningDevicePacket` object with the
+        provided parameters.
+
+        Args:
+            frame_id (integer): the frame ID of the packet.
+            registrant_address (:class:`.XBee64BitAddress`): the 64-bit address of the destination device.
+            options (:class:`.RegisterKeyOptions`): the register options indicating the key source.
+            key (Bytearray): key of the device to register. up to 16 bytes if entering a Link Key or up to
+                18 bytes (16-byte code + 2 byte CRC) if entering an Install Code.
+
+        Raises:
+            ValueError: if ``frame_id`` is less than 0 or greater than 255.
+
+        .. seealso::
+           | :class:`.XBee64BitAddress`
+           | :class:`.XBeeAPIPacket`
+           | :class:`.RegisterKeyOptions`
+        """
+        if frame_id < 0 or frame_id > 255:
+            raise ValueError("Frame id must be between 0 and 255.")
+
+        super().__init__(ApiFrameType.REGISTER_JOINING_DEVICE)
+        self._frame_id = frame_id
+        self.__registrant_address = registrant_address
+        self.__options = options
+        self.__key = key
+
+    @staticmethod
+    def create_packet(raw, operating_mode):
+        """
+        Override method.
+
+        Returns:
+            :class:`.RegisterJoiningDevicePacket`.
+
+        Raises:
+            InvalidPacketException: if the bytearray length is less than 17. (start delim. + length (2 bytes) + frame
+                type + frame id + 64-bit registrant addr. (8 bytes) + 16-bit registrant addr. (2 bytes) + options
+                + checksum = 17 bytes).
+            InvalidPacketException: if the length field of 'raw' is different than its real length. (length field: bytes
+                2 and 3)
+            InvalidPacketException: if the first byte of 'raw' is not the header byte. See :class:`.SpecialByte`.
+            InvalidPacketException: if the calculated checksum is different than the checksum field value (last byte).
+            InvalidPacketException: if the frame type is not :attr:`.ApiFrameType.REGISTER_JOINING_DEVICE`.
+            InvalidOperatingModeException: if ``operating_mode`` is not supported.
+
+        .. seealso::
+           | :meth:`.XBeePacket.create_packet`
+           | :meth:`.XBeeAPIPacket._check_api_packet`
+        """
+        if operating_mode != OperatingMode.ESCAPED_API_MODE and operating_mode != OperatingMode.API_MODE:
```

Comment:
This can be simplified to:
`if operating_mode not in [OperatingMode.ESCAPED_API_MODE, OperatingMode.API_MODE]:`

Reply:
I've changed it in both frames.